lib/measures/tbd/resources/psi.rb

```diff
@@ -147,16 +138,18 @@ def append(p)
     end
   end
 
-  def complete?(s)
-    raise "Unknown '#{s}' PSI set in #{@set}"  unless @set.has_key?(s)
-    raise "'#{s}' rimjoist?"        unless @set[s].has_key?(:rimjoist)
-    raise "'#{s}' parapet?"         unless @set[s].has_key?(:parapet)
-    raise "'#{s}' fenestration?"    unless @set[s].has_key?(:fenestration)
-    raise "'#{s}' concave?"         unless @set[s].has_key?(:concave)
-    raise "'#{s}' convex?"          unless @set[s].has_key?(:convex)
-    raise "'#{s}' balcony?"         unless @set[s].has_key?(:balcony)
-    raise "'#{s}' party?"           unless @set[s].has_key?(:party)
-    raise "'#{s}' grade?"           unless @set[s].has_key?(:grade)
+  def complete?(s) # true/false
+    answer = true
+    answer = false unless @set.has_key?(s)
```

Comment:
I would make each of these:

`answer = (answer && @set.has_key?(s))`

Right now answer could get set to false (missing rimjoist) but then set back to true (has party)